mobile/lib/services/download.service.dart

```diff
@@ -0,0 +1,183 @@
+import 'dart:io';
+
+import 'package:background_downloader/background_downloader.dart';
```

Comment:
Can you rework this to use the repository pattern? With it you should not import this dependency here at all. We could make a new DownloadRepository with methods to download, and add progress/state callbacks.